pkg/gatherers/clusterconfig/gather_monitoring_pvs_test.go

```diff
@@ -0,0 +1,219 @@
+		assertRecord       bool
+		assertError        bool
+	}{
+		{
+			name: "Existent Persistent Volume within the namespace is gathered",
+			pvc: &corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{Name: "mockName", Namespace: "openshift-monitoring"},
+				Spec:       corev1.PersistentVolumeClaimSpec{VolumeName: "test"},
+			},
+			pv: &corev1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{Name: "test"},
+				Status:     corev1.PersistentVolumeStatus{Phase: "Available"},
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{},
+				},
+			},
+			assertErrorsNumber: 0,
+			assertRecordNumber: 1,
+			assertRecord:       true,
+		},
+		{
+			name: "Existent Persistent Volume with unmatching prefix is not gathered",
+			pvc: &corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{Name: "mockFail", Namespace: "openshift-monitoring"},
+				Spec:       corev1.PersistentVolumeClaimSpec{VolumeName: "test"},
+			},
+			pv: &corev1.PersistentVolume{
+				ObjectMeta: metav1.ObjectMeta{Name: "test"},
+				Status:     corev1.PersistentVolumeStatus{Phase: "Available"},
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{},
+				},
+			},
+			assertErrorsNumber: 0,
+			assertRecordNumber: 0,
+		},
+		{
+			name: "Non-existent Persistent Volume within the namespace throws an error",
+			pvc: &corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{Name: "mockName", Namespace: "openshift-monitoring"},
+				Spec:       corev1.PersistentVolumeClaimSpec{VolumeName: "test"},
+			},
+			pv:                 &corev1.PersistentVolume{},
+			assertErrorsNumber: 1,
+			assertRecordNumber: 0,
+			assertError:        true,
+		},
+		{
+			name:               "Non-existent Persistent Volume Claim does not throw any error",
+			pvc:                &corev1.PersistentVolumeClaim{},
+			pv:                 &corev1.PersistentVolume{},
+			assertErrorsNumber: 0,
+			assertRecordNumber: 0,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			// Given
+			coreclient := kubefake.NewSimpleClientset([]runtime.Object{testCase.pvc, testCase.pv}...)
+			gatherer := MonitoringPVGatherer{client: coreclient.CoreV1()}
+
+			// When
+			records, errors := gatherer.gather(context.TODO(), "mockName")
+
+			// Assert
+			assert.Len(t, records, testCase.assertRecordNumber)
+			if testCase.assertRecord {
+				assert.Equal(t, "config/persistentvolumes/test", records[0].Name)
+			}
+
+			assert.Len(t, errors, testCase.assertErrorsNumber)
+			if testCase.assertError {
+				assert.ErrorContains(t, errors[0], "not found")
+			}
+		})
+	}
+}
+
+func Test_GatherMonitoring_unmarshalDefaultPath(t *testing.T) {
+	longValidMock := "prometheusK8s:\n  volumeClaimTemplate:\n    metadata:\n      name: mock\n" +
+		"    spec:\n      storageClassName: local-storage\n      resources:\n        requests:\n          storage: 2Gi\n"
```

Comment:
Just suggestion - maybe we could move this yaml to `testdata` dir as a real yaml...because I guess it might be a little problematic when there's a need to update it in the future